### PR TITLE
scripts: kconfig: Catch warnings generated during symbol evaluation + fix resulting warnings

### PR DESCRIPTION
**Pull request by ulfalizer**
Warnings generated during symbol evaluation were accidentally ignored, due to checking for warnings before writing `.config` and `autoconf.h` (which indirectly evaluates all symbols).

Move the warning checking code to after writing the configuration to catch such warnings. `kconfig.py` still gets rerun if any warnings-turned-errors show up.

Also include a fix for two accidentally swapped `range`s on the `BT_RFCOMM_L2CAP_MTU` symbol. Confusingly, earlier `range`s are still preferred in Zephyr, despite the custom prefer-later-defaults behavior.

### CODE REVIEW

**Comment by ulfalizer** [comment]
CC @carlescufi @jhedberg

File: subsys/bluetooth/host/Kconfig
Comment (cvinayak) [inline subsys/bluetooth/host/Kconfig]: is "range" different from "default", isnt the last evaluation that will be picked? with this change the range will always become BT_RX_BUF_LEN to 32767.

**Review by cvinayak — CHANGES_REQUESTED** [review]

File: subsys/bluetooth/host/Kconfig
Comment (ulfalizer) [reply subsys/bluetooth/host/Kconfig]: @cvinayak 
Yep, it's different, which is confusing. The commit message in https://github.com/zephyrproject-rtos/zephyr/pull/9059/commits/f572976f2c094852af958a6c23a00a2b137793a1 goes over why.

That's one of the reasons that I'm trying to go back to standard Kconfig behavior for defaults. I don't know how many other similar mistakes there are.

Comment (ulfalizer) [reply subsys/bluetooth/host/Kconfig]: See http://docs.zephyrproject.org/porting/board_porting.html#zephyr-specific-kconfig-behavior-for-defaults as well. Only Zephyr does that prefer-later-defaults thing.

That it's inconsistent makes it even nastier. I'm planning to just turn all the defaults around later (along with having an easily-disabled warning about the changed behavior). There's been other gotchas too (for example, `choice`s don't prefer later defaults either).

File: scripts/kconfig/kconfig.py
Comment (cvinayak) [inline scripts/kconfig/kconfig.py]: This is resolved in master now, you dont need this.

**Review by cvinayak — CHANGES_REQUESTED** [review]
minor comment else LGTM

**Comment by ulfalizer** [comment]
@cvinayak 
Removed the whitelist for the fixed issue.

**Review by cvinayak — APPROVED** [review]
LGTM

**Comment by cvinayak** [comment]
@ulfalizer lets wait for another reviewers approval and ask for merging it

**Review by jhedberg — APPROVED** [review]

**Comment by ulfalizer** [comment]
CC @nashif (@carlescufi is on parental leave)

**Review by tyrantlizard77 — APPROVED** [review]

**Review by SebastianBoe — APPROVED** [review]

**Pull request merged**
